app/dao/users_dao.py

```diff
@@ -16,9 +21,7 @@ def save_model_user(usr, update_dict={}, pwd=None):
         usr.password = pwd
         usr.password_changed_at = datetime.utcnow()
     if update_dict:
-        if update_dict.get('id'):
-            del update_dict['id']
-            update_dict.pop('password_changed_at')
+        _remove_values_for_keys_if_present(update_dict, ['id', 'password', 'password_changed_at'])
```

Comment:
Initially it was because the schema checks were being omitted resulting in the password being included in the update_dict. Anyhow I've changed to put it back in and using a separate schema for my changes.